modules/oe_migration_workflow/src/Plugin/migrate/process/SetWorkflowState.php

```diff
@@ -0,0 +1,259 @@
+ * it will be set according to the status property, if the status is set to "1",
+ * then it will be set to the given "published_state", in any other case, it
+ * will be set to the given "unpublished_state".
+ *
+ * Compared to a static mapping, it can also handle cases where the source
+ * entity might not have a moderation state defined (NULL).
+ *
+ * Available configuration keys:
+ *   - workflow_config_name (mandatory): The destination workflow's
+ *     configuration name (string).
+ *     Default value: "workflows.workflow.oe_corporate_workflow".
+ *   - published_state (mandatory): The destination workflow's published state
+ *     (string).
+ *     Default value: "published".
+ *   - unpublished_state (mandatory): The destination workflow's unpublished
+ *     state (string).
+ *     Default value: "draft".
+ *
+ * @codingStandardsIgnoreStart
+ * Example with default configuration:
+ * @code
+ * process:
+ *   moderation_state:
+ *    - plugin: oe_migration_set_workflow_state
+ *      source: moderation_state
+ * @endcode
+ *
+ * Example with user defined configuration:
+ * @code
+ * process:
+ *   moderation_state:
+ *    - plugin: oe_migration_set_workflow_state
+ *      source: moderation_state
+ *      workflow_config_name: workflows.workflow.oe_corporate_workflow
+ *      published_state: published
+ *      unpublished_state: draft
+ * @endcode
+ * @codingStandardsIgnoreEnd
+ *
+ * @MigrateProcessPlugin(
+ *   id = "oe_migration_set_workflow_state"
+ * )
+ */
+class SetWorkflowState extends ProcessPluginBase implements ContainerFactoryPluginInterface, ValidConfigurableMigrationPluginInterface {
+
+  /**
+   * The configuration manager.
+   *
+   * @var \Drupal\Core\Config\ConfigManagerInterface
+   */
+  protected $configManager;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\migrate\MigrateException
+   */
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    ConfigManagerInterface $config_manager,
+    EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    // The default configuration.
+    $this->configuration += [
+      'workflow_config_name' => 'workflows.workflow.oe_corporate_workflow',
+      'published_state' => 'published',
+      'unpublished_state' => 'draft',
+    ];
+    $this->configManager = $config_manager;
+    $this->entityTypeManager = $entity_type_manager;
+    // Check if the configuration is set correctly.
+    $this->validateConfigurationKeys();
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
```

Comment:
Done

Reply:
Two core process plugins (MigrationLookup and MakeUniqueBase) declare the exception, some others don't. I guess that there is not a global rule about that and it relies on coding styles. I personally prefer to provide as much information as possible in the comments of a method and declare the exceptions :)

Reply:
You're right Ioannis, Maybe it's not an error. I changed it because I didn't know the rule and I just tried to resolve al comments in order to allow the merge. So, please, let me know if I have to change back it. :)

Reply:
We do not add @throws in openeuropa components.